os/20211014-s390x-secure-execution.md

```diff
@@ -0,0 +1,215 @@
+# IBM Secure Execution for Linux
+---
+
+# Theory of operation
+
+[IBM Secure Execution for Linux®](https://www.ibm.com/docs/en/linux-on-systems?topic=virtualization-introducing-secure-execution-linux) is a hardware-based security technology that is
+built into the IBM z15™ and LinuxONE III generation systems. It is designed to
+provide scalable isolation for individual workloads to help protect them from not
+only external attacks, but also insider threats. In traditional x86 ring architectures,
+the host can access the memory and data of guest applications freely, leading to the
+potential for malicious software to be proliferated throughout the entire system.
+Isolation between host and guest environments is necessary to help prevent system compromise.
+Users can enable Secure Execution feature on IBM zKVM Host and Guest VMs to isolate them from each other.
+
+That means:
+1. An encrypted Linux image, called `sd-boot`, (kernel + ramdisk + cmdline) is created using the host public key and a customer-specific key. Since the encryption keys are stored on the IBM Z hardware and firmware, the encrypted image can only be executed in a virtual machine on the host(s) it has been prepared for, and the image can’t be decrypted or tampered with outside of the designated host(s).
+2. zKVM host cannot access memory of secured guest VM
+3. Guest VMs cannot access memory of other secured guest VMs
+4. If anyone tries to access memory of secured guest VM, than this gets blocked on HW level
+
+## Specific attacks being addressed
+- protect guests against the system
+- protect guests against a Linux Admin (on the host)
+- protect the data services use within a secured guests from host admin and/or other guests
+
+## Specific attacks not being addressed
+- this does not attempt to secure initial system provisioning (Ignition)
```

Comment:
Just to be clear - you say that firstboot will be done without SE? It's not obvious to me why that's ok / desired.

My expectation would have been that the user either manually turns the fcos install artifacts into an SE bootimage, or coreos-installer (?) learns how to build the SE bootimage for installation.